src/main/java/com/sourcery/gymapp/backend/workout/repository/WorkoutRepository.java

```diff
@@ -16,4 +19,57 @@ public interface WorkoutRepository extends JpaRepository<Workout, UUID> {
 
     List<Workout> findByUserIdAndDateBetween(UUID userId, ZonedDateTime startDate,
                                              ZonedDateTime endDate, Sort sort);
+
+    @Query(
+            "SELECT COUNT(w) " +
+            "FROM Workout w " +
+            "WHERE (w.userId = :currentUserId) " +
+            "AND (w.date BETWEEN :startOfTheMonth AND :endOfTheMonth)"
+    )
+    int countWorkoutsByUserIdAndDateBetween(UUID currentUserId,
+                                            ZonedDateTime startOfTheMonth,
+                                            ZonedDateTime endOfTheMonth);
+
+    @Query(
+            "SELECT SUM(wes.setNumber * wes.reps * wes.weight) as total_weight, w.userId " +
+            "FROM WorkoutExerciseSet wes " +
+            "LEFT JOIN WorkoutExercise we " +
+            "ON wes.workoutExercise.id = we.id " +
+            "RIGHT JOIN Workout w " +
+            "ON we.workout.id = w.id " +
+            "WHERE (w.userId = :currentUserId) " +
+            "AND (w.date BETWEEN :startOfTheMonth AND :endOfTheMonth)" +
+            "GROUP BY w.userId"
+    )
+    Integer getTotalWeightByUserIdAndDateBetween(UUID currentUserId,
```

Comment:
or instead of optional<integer> (since we always need to have a number on service layer) we can use COALESCE(SUM(...), 0) to always return non-null results, which could be more useful